cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -220,6 +264,51 @@ ENUMERIC_TYPES_UNARY(DEGREES, float64)
   }
 POWER(float64, float64, float64)
 
+// Sqrt
+#define SQRT(IN_TYPE, OUT_TYPE)                                             \
+  FORCE_INLINE                                                              \
+  gdv_##OUT_TYPE sqrt_##IN_TYPE(gdv_##IN_TYPE in) {                         \
+    return static_cast<gdv_##OUT_TYPE>(sqrt(static_cast<long double>(in))); \
+  }
+ENUMERIC_TYPES_UNARY(SQRT, float64)
+
+// Sign
+#define SIGN(IN_TYPE)                                                               \
+  FORCE_INLINE                                                                      \
+  gdv_##IN_TYPE sign_##IN_TYPE(gdv_##IN_TYPE in) {                                  \
+    if (in == 0.0) return in;                                                       \
+    return static_cast<gdv_##IN_TYPE>(copysign(1.0, static_cast<long double>(in))); \
+  }
+ENUMERIC_TYPES(SIGN)
+
+// Lshift
+#define LSHIFT(IN_TYPE)                                                  \
+  FORCE_INLINE                                                           \
+  gdv_##IN_TYPE lshift_##IN_TYPE(gdv_##IN_TYPE in1, gdv_##IN_TYPE in2) { \
+    return static_cast<gdv_##IN_TYPE>(in1 << in2);                       \
+  }
+LSHIFT(int32)
+LSHIFT(int64)
+LSHIFT(uint32)
+LSHIFT(uint64)
+
+// Lshift
```

Comment:
Lshift -> Rshift

Reply:
It would be good to write the definition of Rshift. Shift logical or shift arithmetic?

Reply:
The definition was added